### PR TITLE
HDDS-5285. Document block deleting service classes

### PR DESCRIPTION
**Pull request by sky76093016**
## What changes were proposed in this pull request?

Add class-level javadoc for DeletedBlockLogStateManager and DeletedBlockLogStateManagerImpl.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/projects/HDDS/issues/HDDS-5285

## How was this patch tested?

No test.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManager.java [hunk 1]
```diff
@@ -27,7 +27,8 @@
 import java.util.ArrayList;
 
 /**
- * TODO: javadoc.
+ * DeletedBlockLogStateManager interface to provide BlocksTransaction
```
Comment (jojochuang) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManager.java:30]: I think we can make this more verbose.

if you trace the code, you'll find it ultimately writes to SCMMetadataStore.getDeletedBlocksTXTable() which says "A Table that keeps the deleted blocks lists and transactions."

You might say it's an interface to manage deleted blocks and record them in the underlying persist store.

Comment (sky76093016) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManager.java:30]: This seems to be a good proposal, let me revise it.

**Review by cku328 — APPROVED** [review]
+1 Thanks @sky76093016 for the update.

**Review by jojochuang — APPROVED** [review]

**Comment by cku328** [comment]
PR merged. 
Thanks @sky76093016  for the contribution and @jojochuang for the review.

**Pull request merged**